proto/api/v1/image_service.proto

```diff
@@ -4,12 +4,13 @@ option go_package = "v1";
 option java_package = "io.stackrox.proto.api.v1";
 
 import weak "google/api/annotations.proto";
-import "api/v1/search_service.proto";
 import "api/v1/empty.proto";
+import "api/v1/search_service.proto";
+import "internalapi/scanner/v4/common.proto";
```

Comment:
I know about the point you are raising, actually. You haven't talked about alternatives, but here are the options I thought when deciding about this:

1. Make scanner.v4 part of the public API.
2. Make only parts of scanner.v4 the public API.
3. Consider some APIs exposed by Central have semantics to the internal one, even though they are part of the "api/v1" definition.

I chose option 3 because making sure that "EnrichLocalImageInternal" and its associated types have public-facing semantics would make this change more complex and increase its impact. In the end, it could be more confusing to enforce scanner.v4's internal types to adopt public semantics, either for all of the protos or only some of them.

Curiously, the API carries the name "Internal" to it. It might mean it was designed to be internal.

Let me know if you have some other alternative we should consider.

Reply:
Another alternative is to maintain a copy of the protos for internal vs external, although they may be the same now, should the APIs need to digress in the future can do so. This was done for the delegated registry service/config based on [PR review](https://github.com/stackrox/stackrox/pull/5951#discussion_r1188330096) and offline discussions - leading to separate protos for storage, internal api, and external api, and code to [convert between them](https://github.com/stackrox/stackrox/blob/master/central/delegatedregistryconfig/convert/convert.go).

Reply:
@dcaravel Based on the discussion, it seems like the leaked proto came from storage. These protos are used for the database models, which may need changes that could affect the APIs negatively. For instance, denormalizing fields, migrating fields, adding metadata fields, and so on. However, the types under "internalapi.scanner.v4" are likely to remain unchanged as the Scanner V4/ClairCore types develop internally. That's why we've already converted ClairCore types to this internal API type. Basically, there isn't a strong reason to change the internal Scanner V4 types with changes that we don't want to expose or transmit to Central Scanner.

So, it's possible that the intermediary type used for `EnrichLocalImageInternal` would need modification anyway, right? And since EnrichLocalImageInternal is internal as far as I know, this seems acceptable. While it might be safer to separate them, my question is whether it's worth the extra effort of creating separate protos and converters for these non-storage types.

Reply:
Sounds OK to leave as is for now and if we need to in the future separate the proto's then. Marking discussion resolved.

Reply:
From what I can tell, this is just used for an "internal" endpoint (not sure why we have been putting these in the external API path, but whatever. This original decision outdates all of us I think), so I think this is ok, as long as the "internalapi" stuff isn't exposed to users, which I don't think is the case